admin/spec/helpers/solidus_admin/layout_helper_spec.rb

```diff
@@ -3,6 +3,14 @@
 require "spec_helper"
 
 RSpec.describe SolidusAdmin::LayoutHelper, :helper do
+  describe '#current_locale' do
+    it "returns the current locale" do
+      expect(
+        helper.current_locale(backend: double(locale: :en))
```

Comment:
With a simple `double` RSpec won't apply spies to check if the original method exist, that would make it more robust. 

To me would also be fine if we stubbed directly on `I18n`, is there any utility in injecting the constant other than testing?

Reply:
I'm not worried about the test not checking the original interface, which will be fully covered at the integration layer, and that's not something the unit level should worry about. I like this pattern where we limit hidden pre-conditions for a method to its parameter defaults. However, if you feel strongly about it, I won't die because of doing it the other way :slightly_smiling_face:

Reply:
I don't feel strongly at all, but I'm interested un understanding it more deeply, please stay alive and keep it ❤️ 

I guess the things that confuse me the most are: 
- what would be the use-case for passing a different thing as the i18n backend? For example it's fairly easy to `allow(I18n).to receive(:locale).and_respond(:en)` without all this harness.
- what are the hidden preconditions in this context? 
- why the unit level shouldn't worry that its default value meet its expectations? 


_Thanks in advance for your patience!_

Reply:
> please stay alive and keep it

I'll [stay alive](https://www.youtube.com/watch?v=fNFzfwLM72c) anyway! :laughing:

As with so many things, that's just opinionated stuff:

> what would be the use-case for passing a different thing as the i18n backend? For example it's fairly easy to allow(I18n).to receive(:locale).and_respond(:en) without all this harness

In this case, it's only useful for testing purposes. I'm not fond of stubbing, and I try to avoid `allow... to receive...` and leave it for some integration cases. In my view, it's too tied to the method implementation, and I prefer providing external interfaces.

> what are the hidden preconditions in this context?

The hidden precondition is the I18n module. The method implementation expects it to be there and to have been configured. Of course, Ruby is not pure, but we can limit those pre-conditions to the method signature. It helps me reason about methods.

> why the unit level shouldn't worry that its default value meet its expectations?

In my view, it should worry about the interface that its collaborator implements, but not about its nature.

Again, I think there can be different views about this stuff and none is the right or wrong one :slightly_smiling_face:

Reply:
Thanks, this helps me a lot understanding the reasoning behind this style 👍👍👍